### PR TITLE
 libusb: bump to 1.0.24

### PR DESCRIPTION
**Pull request by kenmays**
Passes all unit tests on Haiku R1B2 x86_64.

### CODE REVIEW

**Comment by pulkomandy** [comment]
Have you confirmed that the patch has been upstreamed and is not needed anymore, or did you ignore it because newer version just compiles without it?

File: dev-libs/libusb/libusb-1.0.24.recipe [hunk 1]
```diff
@@ -22,14 +22,14 @@ COPYRIGHT="2001 Johannes Erdfelt
 LICENSE="GNU LGPL v2.1"
 REVISION="1"
 SOURCE_URI="https://github.com/libusb/libusb/archive/v$portVersion.tar.gz"
-CHECKSUM_SHA256="02620708c4eea7e736240a623b0b156650c39bfa93a14bcfa5f3e05270313eba"
+CHECKSUM_SHA256="b7724c272dfc5713dce88ff717efd60f021ca5b7c8e30f08ebb2c42d2eea08ae"
 SOURCE_FILENAME="libusb-$portVersion.tar.gz"
-PATCHES="libusb-$portVersion.patchset"
+#PATCHES="libusb-$portVersion.patchset"
```
Comment (extrowerk) [inline dev-libs/libusb/libusb-1.0.24.recipe:27]: Unneded

Comment (kenmays) [reply dev-libs/libusb/libusb-1.0.24.recipe:27]: I verified that all prior Haiku-related libusb patches were integrated in this release.

**Comment by kenmays** [comment]
All prior Haiku libusb patches integrated in this release.

**Comment by Vidrep** [comment]
Successfully built package on 64 bit nightly. Tested with 2 different USB scanners, OK.

**Comment by Begasus** [comment]
Build fails with gcc2 (needs compiler with C11 support)

File: dev-libs/libusb/libusb-1.0.24.recipe (hunk 1 above)
Comment (Begasus) [reply dev-libs/libusb/libusb-1.0.24.recipe:27]: Having a quick look on their github page seems to confirm, no need for it then:
```diff
- #PATCHES="libusb-$portVersion.patchset"
```

**Comment by Begasus** [comment]
@kenmays quite some recipes deppend on libusb_1.0, some even for the x86_gcc2 version
For now maybe re-enable version 1.0.22 for x86_gcc2 and revbump others that need the newer version
```
gnupg-1.4.23-3-x86_gcc2.hpkg:
	requires "lib:libusb_1.0 >= 0.1.0" of package "libusb_compat-0.1.5-7" could not be resolved
	requires "lib:libusb_0.1 >= 0.1.5" of package "gnupg-1.4.23-3" could not be resolved

libusb_compat_devel-0.1.5-7-x86_gcc2.hpkg:
	requires "libusb_compat == 0.1.5" of package "libusb_compat_devel-0.1.5-7" could not be resolved

libyubikey_devel-1.13-5-x86_gcc2.hpkg:
	requires "lib:libusb_1.0 >= 0.1.0" of package "libyubikey-1.13-5" could not be resolved
	requires "libyubikey == 1.13" of package "libyubikey_devel-1.13-5" could not be resolved

libyubikey-1.13-5-x86_gcc2.hpkg:
	requires "lib:libusb_1.0 >= 0.1.0" of package "libyubikey-1.13-5" could not be resolved

dfu_util-0.8-1-x86_gcc2.hpkg:
	requires "lib:libusb_1.0 >= 0.1.0" of package "dfu_util-0.8-1" could not be resolved

libusb_compat-0.1.5-7-x86_gcc2.hpkg:
	requires "lib:libusb_1.0 >= 0.1.0" of package "libusb_compat-0.1.5-7" could not be resolved
```

**Pull request merged**